<launch>
  <arg name="model" default="$(find r2d2)/urdf/r2d2_two_arms.urdf.xacro"/>

  <include file="$(find r2d2)/launch/gazebo.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <rosparam command="load"
            file="$(find r2d2)/config/joints.yaml"
            ns="r2d2_joint_state_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/head.yaml"
            ns="r2d2_head_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/gripper.yaml"
            ns="r2d2_gripper_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/left_arm.yaml"
            ns="r2d2_left_arm_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/right_arm.yaml"
            ns="r2d2_right_arm_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/diffdrive.yaml"
            ns="r2d2_diff_drive_controller" />

  <node name="r2d2_controller_spawner" pkg="controller_manager" type="spawner"
    args="r2d2_joint_state_controller
          r2d2_head_controller
          r2d2_gripper_controller
          r2d2_left_arm_controller
          r2d2_right_arm_controller
          r2d2_diff_drive_controller
          --shutdown-timeout 3"/>

</launch>
